projects/client/RabbitMQ.Client/src/client/api/IModel.cs

```diff
@@ -552,4 +557,8 @@ public interface IModel : IDisposable
         /// </summary>
         TimeSpan ContinuationTimeout { get; set; }
     }
+    public class BatchMessage{
```

Comment:
Stores the variable data between publish invocations using the BasicPublish, which are the BasicProperties and the body of bytes.
The name I chose is BatchMessage; it’s a single message as part of the batch.
The batch is an IEnumerable<BatchMessage> that will be compiled into a single stream and sent on the socket.
It would be possible to change this to Batch, where we create another class that is simply 
`public class Batch : List<BatchMessage>{}`

Reply:
As owner @michaelklishin of the source please consider this, 

A way to avoid batching and gain the performance is to buffer outbound frames for sending on a seperate thread.

This however causes behavioural changes in the client, so I chose to write Batch functionality instead.  Plus, this fits well for many of the use-cases for RabbitMQ.

Interestingly though, using the oubound frame buffer and thread for sending, you get benifits with Ack's being grouped and sent as a single stream on the socket too, which increases read throughput.  To achieve this however, I took a dependancy on BufferBlock<T> from TPL DataFlow.

I don't believe BufferStream is usefull for writes because it doesn't automatically flush periodically.
A new bepoke stream may though, with a one/two/x second timer to flush any remaining data.
But again, this may have similar behavioural issues.